ttml.go

```diff
@@ -422,13 +507,186 @@ func ReadFromTTML(i io.Reader) (o *Subtitles, err error) {
 
 		}
 		s.Lines = append(s.Lines, *l)
-
+		//setting final alignment of entire <p>
+		if finalAlignment != nil {
+			ts.TTMLInStyleAttributes.TextAlign = finalAlignment
+		}
+		if finalExtent != nil {
+			ts.TTMLInStyleAttributes.Extent = finalExtent
+		}
+		if finalOrigin != nil {
+			ts.TTMLInStyleAttributes.Origin = finalOrigin
+		}
+		if finalDisplayAlignment != nil {
+			ts.TTMLInStyleAttributes.DisplayAlign = finalDisplayAlignment
+		}
+		if finalPadding != nil {
+			ts.TTMLInStyleAttributes.Padding = finalPadding
+		}
+		if finalWritingMode != nil {
+			ts.TTMLInStyleAttributes.WritingMode = finalWritingMode
+		}
+		s.InlineStyle = ts.TTMLInStyleAttributes.styleAttributes()
 		// Append subtitle
 		o.Items = append(o.Items, s)
 	}
+
+	//Pushing Inline style from parent tag to every child tag then combining it with child tag's Style to create a new style and then assigning that style to the child tag.
+	customStyleNumber := 1
+	var customStyleMap = make(map[Style]string)
```

Comment:
I feel it should be the other way around and with a pointer `var customStyleMap = make(map[string]*Style)`, and you should rename this variables to `customStyles`